lightkurve/seismology.py

```diff
@@ -0,0 +1,908 @@
+
+    def plot(self, **kwargs):
+        return self.periodogram.plot(**kwargs)
+
+
+    def _get_fwhm(self, numax):
+        """In a power spectrum of a solar-like oscillator, the power of the
+        modes of oscillation will appear in the shape of that looks
+        approximately Gaussian, for all basic purposes, also referred to as the
+        'mode envelope'. For a given numax (the central frequency of the mode
+        envelope), the expected Full Width Half Maximum of the envelope is known
+        as a function of numax for evolved Red Giant Branch stars as follows
+        (see Mosser et al 2010):
+
+        fwhm = 0.66 * numax^0.88 .
+
+        If the maximum frequency in the periodogram is less than 500 microhertz,
+        this function will default to the above equation under the assumption it
+        is dealing with an RGB star, which oscillate at lower frequencies.
+
+        If the maximum frequency is above 500 microhertz, the envelope is given
+        as a different function of numax (see Lund et al. 2017), as
+
+        fwhm = 0.25 * numax,
+
+        in which case the function assumes it is dealing with a main sequence
+        star, which oscillate at higher frequencies.
+
+        Parameters:
+        ----------
+            numax : float
+                The estimated position of the numax of the power spectrum. This
+                is used to calculated the region autocorrelated with itself.
+
+        Returns:
+        --------
+            fwhm: float
+                The estimate full-width-half-maximum of the seismic mode envelope
+        """
+        #Calculate the index FWHM for a given numax
+        if u.Quantity(self.frequency[-1], u.microhertz) > u.Quantity(500., u.microhertz):
+            fwhm = 0.25 * numax
+        else:
+            fwhm = 0.66 * numax**0.88
+        return fwhm
+
+
+    def _autocorrelate(self, numax, window=25., frequency_spacing=None):
+        """An autocorrelation function (ACF) for seismic mode envelopes.
+        We autocorrelate a region with a width of `window` (in microhertz)
+        around a central frequency `numax` (in microhertz). The window size is
+        determined based on the location of the nyquist frequency when
+        estimating numax, and based on the expected width of the mode envelope
+        of the asteroseismic oscillations when calculating deltanu.
+
+        Parameters:
+        ----------
+            numax : float
+                The estimated position of the numax of the power spectrum. This
+                is used to calculated the region autocorrelated with itself.
+
+            window : int or float
+                The width of the autocorrelation window around the central
+                frequency numax.
+
+            frequency_spacing : float
+                The frequency spacing of the periodogram. If none is passed, it
+                is calculated internally. This should never be set by the user.
```

Comment:
If `frequency_spacing` should never be set by the user, is there a use-case for needing this argument?